core/adapters/eth_tx.go

```diff
@@ -140,8 +140,14 @@ func (e *EthTx) checkEthTxForReceipt(ethTxID int64, input models.RunInput, s *st
 		return models.NewRunOutputPendingOutgoingConfirmationsWithData(input.Data())
 	}
 
-	output := models.JSON{}
-	output, err = output.Add("result", (*hash).Hex())
+	hexHash := (*hash).Hex()
+
+	output := input.Data()
+	output, err = output.MultiAdd(models.KV{
+		"result": hexHash,
```

Comment:
I assume `result` and `latestOutgoingTxHash` are needed by the UI?

Reply:
`result` is a special magical value that is passed through the chain of tasks.

`latestOutgoingTxHash` is ONLY required by the UI, I left a comment to hopefully make this clear.